diff --git a/tests/basictest.cpp b/tests/basictest.cpp
index dc11752..b81b753 100644
--- a/tests/basictest.cpp
+++ b/tests/basictest.cpp
@@ -2145,7 +2145,7 @@ TEST_CASE("integer_times_pow10") {
   verify_integer_multiplication_by_power_of_10<unsigned long long>(
       3141592653589793238, -18, 3.141592653589793238);
 
-  for (int mode : {FE_UPWARD, FE_DOWNWARD, FE_TOWARDZERO, FE_TONEAREST}) {
+  for (int mode : {(int)FE_UPWARD, (int)FE_DOWNWARD, (int)FE_TOWARDZERO, (int)FE_TONEAREST}) {
     fesetround(mode);
     INFO("fesetround(): " << std::string{round_name(mode)});
 
