diff -Naur --exclude build llvm-project-qnx-22.1.4_p1.orig/lld/ELF/Config.h llvm-project-qnx-22.1.4_p1/lld/ELF/Config.h
--- lld.orig/ELF/Config.h	2026-05-15 17:40:49.000000000 +0000
+++ lld/ELF/Config.h	2026-07-31 23:26:30.000000000 +0000
@@ -601,6 +601,8 @@
   std::unique_ptr<StringTableSection> strTab;
   std::unique_ptr<SymbolTableBaseSection> symTab;
   std::unique_ptr<SymtabShndxSection> symTabShndx;
+  // QNX
+  std::unique_ptr<SyntheticSection> qnxStack;
 };
 
 struct Ctx : CommonLinkerContext {
diff -Naur --exclude build llvm-project-qnx-22.1.4_p1.orig/lld/ELF/Driver.cpp llvm-project-qnx-22.1.4_p1/lld/ELF/Driver.cpp
--- lld.orig/ELF/Driver.cpp	2026-05-15 17:40:49.000000000 +0000
+++ lld/ELF/Driver.cpp	2026-08-05 07:05:22.000000000 +0000
@@ -1553,7 +1553,9 @@
   if (args.hasArg(OPT_randomize_section_padding))
     ctx.arg.randomizeSectionPadding =
         args::getInteger(args, OPT_randomize_section_padding, 0);
-  ctx.arg.singleRoRx = !args.hasFlag(OPT_rosegment, OPT_no_rosegment, true);
+  // TODO: figure out a "correct" way to do this that is upstreamable
+  // QNX has a bug in its elfloader where you MUST have a singleRoRx
+  ctx.arg.singleRoRx = !args.hasFlag(OPT_rosegment, OPT_no_rosegment, ctx.arg.dynamicLinker.contains("ldqnx") ? false : true);
   ctx.arg.singleXoRx = !args.hasFlag(OPT_xosegment, OPT_no_xosegment, false);
   ctx.arg.soName = args.getLastArgValue(OPT_soname);
   ctx.arg.sortSection = getSortSection(ctx, args);
diff -Naur --exclude build llvm-project-qnx-22.1.4_p1.orig/lld/ELF/InputFiles.cpp llvm-project-qnx-22.1.4_p1/lld/ELF/InputFiles.cpp
--- lld.orig/ELF/InputFiles.cpp	2026-05-15 17:40:49.000000000 +0000
+++ lld/ELF/InputFiles.cpp	2026-08-05 06:47:15.000000000 +0000
@@ -1084,6 +1084,45 @@
 }
 
 template <class ELFT>
+static bool isQnxStack(Ctx &ctx, const InputSection &sec) {
+  using Elf_Nhdr = typename ELFT::Nhdr;
+  using Elf_Note = typename ELFT::Note;
+
+  constexpr uint32_t QNX_NOTE_STACK = 3; // Target note type
+
+  ArrayRef<uint8_t> data = sec.content();
+  auto err = [&](const uint8_t *place) -> ELFSyncStream {
+    auto diag = Err(ctx);
+    diag << sec.file << ":(" << sec.name << "+0x"
+         << Twine::utohexstr(place - sec.content().data()) << "): ";
+    return diag;
+  };
+
+  while (!data.empty()) {
+    if (data.size() < sizeof(Elf_Nhdr)) {
+      return false;
+    }
+
+    auto *nhdr = reinterpret_cast<const Elf_Nhdr *>(data.data());
+    size_t noteSize = nhdr->getSize(sec.addralign);
+
+    if (data.size() < noteSize) {
+      err(data.data()) << "corrupt note section: note header claims size larger than section bounds";
+      return false;
+    }
+
+    Elf_Note note(*nhdr);
+    if (nhdr->n_type == QNX_NOTE_STACK && note.getName() == "QNX")
+      return true;
+
+    // Advance to the next note record safely using alignment rules
+    data = data.slice(noteSize);
+  }
+
+  return false;
+}
+
+template <class ELFT>
 InputSectionBase *ObjFile<ELFT>::getRelocTarget(uint32_t idx, uint32_t info) {
   if (info < this->sections.size()) {
     InputSectionBase *target = this->sections[info];
@@ -1177,6 +1217,10 @@
     // against it.
     if (name == ".note.gnu.build-id")
       return &InputSection::discarded;
+
+    if ((name == ".note" || name == ".note.qnx.stack") && isQnxStack<ELFT>(ctx, InputSection(*this, sec, name))) {
+      return &InputSection::discarded;
+    }
   }
 
   // The linker merges EH (exception handling) frames and creates a
diff -Naur --exclude build llvm-project-qnx-22.1.4_p1.orig/lld/ELF/SyntheticSections.cpp llvm-project-qnx-22.1.4_p1/lld/ELF/SyntheticSections.cpp
--- lld.orig/ELF/SyntheticSections.cpp	2026-05-15 17:40:49.000000000 +0000
+++ lld/ELF/SyntheticSections.cpp	2026-07-31 23:47:16.000000000 +0000
@@ -4566,6 +4566,26 @@
          alignTo(ctx.arg.packageMetadata.size() + 1, 4);
 }
 
+QNXStackNote::QNXStackNote(Ctx &ctx)
+    : SyntheticSection(ctx, ".note.qnx.stack", SHT_NOTE, SHF_ALLOC, 4) {
+    
+    stackSize = ctx.arg.zStackSize ? ctx.arg.zStackSize : 0;
+    stackAlloc = stackSize;
+    flags = (ctx.arg.zGnustack == GnuStackKind::Exec) ? 0 : 1;
+}
+
+void QNXStackNote::writeTo(uint8_t *buf) {
+    write32(ctx, buf + 0, 4);
+    write32(ctx, buf + 4, 12);
+    write32(ctx, buf + 8, 3);  // QNT_STACK (3)
+
+    memcpy(buf + 12, "QNX", 4);
+
+    write32(ctx, buf + 16, stackSize);
+    write32(ctx, buf + 20, stackAlloc);
+    write32(ctx, buf + 24, flags);
+}
+
 // Helper function, return the size of the ULEB128 for 'v', optionally writing
 // it to `*(buf + offset)` if `buf` is non-null.
 static size_t computeOrWriteULEB128(uint64_t v, uint8_t *buf, size_t offset) {
@@ -4930,6 +4950,11 @@
     add(*ctx.in.gnuStack);
   }
 
+  if (ctx.arg.dynamicLinker.contains("ldqnx") && (ctx.arg.zStackSize || (ctx.arg.zGnustack != GnuStackKind::None))){
+    ctx.in.qnxStack = std::make_unique<QNXStackNote>(ctx);
+    add(*ctx.in.qnxStack);
+  }
+
   if (ctx.in.symTab)
     add(*ctx.in.symTab);
   if (ctx.in.symTabShndx)
diff -Naur --exclude build llvm-project-qnx-22.1.4_p1.orig/lld/ELF/SyntheticSections.h llvm-project-qnx-22.1.4_p1/lld/ELF/SyntheticSections.h
--- lld.orig/ELF/SyntheticSections.h	2026-05-15 17:40:49.000000000 +0000
+++ lld/ELF/SyntheticSections.h	2026-07-31 23:37:02.000000000 +0000
@@ -1439,6 +1439,18 @@
   SmallVector<const Symbol *, 0> symbols;
 };
 
+// .note.GNU-stack section.
+class QNXStackNote : public SyntheticSection {
+public:
+  QNXStackNote(Ctx &ctx);
+  void writeTo(uint8_t *buf) override;
+  size_t getSize() const override { return 28; }
+private:
+    uint32_t stackSize;
+    uint32_t stackAlloc;
+    uint32_t flags;
+};
+
 template <class ELFT> void createSyntheticSections(Ctx &);
 InputSection *createInterpSection(Ctx &);
 MergeInputSection *createCommentSection(Ctx &);
