--- abuild-3.15.0/default.conf.orig	2025-12-18 00:15:46.000000000 +0000
+++ abuild-3.15.0/default.conf	2025-12-18 00:16:03.000000000 +0000
@@ -1,6 +1,6 @@
 export CFLAGS="-Os -fstack-clash-protection -Wformat -Werror=format-security"
 export CXXFLAGS="-Os -fstack-clash-protection -Wformat -Werror=format-security"
-export CXXFLAGS="$CXXFLAGS -D_GLIBCXX_ASSERTIONS=1 -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS=1 -D_LIBCPP_ENABLE_HARDENED_MODE=1"
+export CXXFLAGS="$CXXFLAGS -D_GLIBCXX_ASSERTIONS=1 -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS=1 -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST"
 export LDFLAGS="-Wl,--as-needed,-O1,--sort-common"
 export GOFLAGS="-buildmode=pie -modcacherw -trimpath -buildvcs=false"
 # Do note that these should work with at least GDC and LDC
