diff --git a/json-glib/tests/parser.c b/json-glib/tests/parser.c
index f924d52..6946e88 100644
--- a/json-glib/tests/parser.c
+++ b/json-glib/tests/parser.c
@@ -302,9 +302,15 @@ test_simple_array (void)
 
       g_assert_cmpint (json_array_get_length (array), ==, test_simple_arrays[i].len);
 
+      const char* type_name = g_type_name (test_simple_arrays[i].gtype);
+      if(type_name == NULL)
+      {
+        type_name = "(null)";
+      }
+
       g_test_message ("checking element %d is of the desired type %s...",
                       test_simple_arrays[i].element,
-                      g_type_name (test_simple_arrays[i].gtype));
+                      type_name);
 
       JsonNode *node = json_array_get_element (array, test_simple_arrays[i].element);
       g_assert_nonnull (node);
@@ -397,9 +403,15 @@ test_simple_object (void)
                       test_simple_objects[i].size);
       g_assert_cmpint (json_object_get_size (object), ==, test_simple_objects[i].size);
 
+      const char* type_name = g_type_name (test_simple_objects[i].gtype);
+      if(type_name == NULL)
+      {
+        type_name = "(null)";
+      }
+
       g_test_message ("checking member '%s' exists and is of the desired type '%s'...",
                       test_simple_objects[i].member,
-                      g_type_name (test_simple_objects[i].gtype));
+                      type_name);
       JsonNode *node = json_object_get_member (object, test_simple_objects[i].member);
       g_assert_nonnull (node);
       g_assert_true (json_node_get_parent (node) == root);
