diff --git a/src/common-image.c b/src/common-image.c
index 7059bf4..a505314 100644
--- a/src/common-image.c
+++ b/src/common-image.c
@@ -25,6 +25,14 @@
 
 #include "common-image.h"
 
+static void *_caca_alloc2d(size_t width, size_t height, size_t elem_size)
+{
+    if (width == 0 || height == 0 || elem_size == 0 || SIZE_MAX / width / height < elem_size)
+        return NULL;
+    return malloc(width * height * elem_size);
+}
+
+
 #if !defined(USE_IMLIB2)
 static unsigned int u32fread(caca_file_t *);
 static unsigned int u16fread(caca_file_t *);
