diff --git a/Source/JavaScriptCore/runtime/StructureID.h b/Source/JavaScriptCore/runtime/StructureID.h
index 2f3eda3e..98faea69 100644
--- a/Source/JavaScriptCore/runtime/StructureID.h
+++ b/Source/JavaScriptCore/runtime/StructureID.h
@@ -46,7 +46,7 @@ static_assert(MACH_VM_MAX_ADDRESS_RAW == MACH_VM_MAX_ADDRESS);
 #if !ENABLE(STRUCTURE_ID_WITH_SHIFT)
 #if defined(STRUCTURE_HEAP_ADDRESS_SIZE_IN_MB) && STRUCTURE_HEAP_ADDRESS_SIZE_IN_MB > 0
 constexpr uintptr_t structureHeapAddressSize = STRUCTURE_HEAP_ADDRESS_SIZE_IN_MB * MB;
-#elif PLATFORM(PLAYSTATION)
+#elif PLATFORM(PLAYSTATION) || OS(QNX)
 constexpr uintptr_t structureHeapAddressSize = 128 * MB;
 #elif PLATFORM(IOS_FAMILY) && CPU(ARM64) && !CPU(ARM64E)
 constexpr uintptr_t structureHeapAddressSize = 512 * MB;
diff --git a/Source/WTF/wtf/NumberOfCores.cpp b/Source/WTF/wtf/NumberOfCores.cpp
index 03d79f26..c75123b8 100644
--- a/Source/WTF/wtf/NumberOfCores.cpp
+++ b/Source/WTF/wtf/NumberOfCores.cpp
@@ -34,6 +34,8 @@
 #include <sys/sysctl.h>
 #elif OS(LINUX) || OS(AIX) || OS(OPENBSD) || OS(NETBSD) || OS(FREEBSD) || OS(HAIKU)
 #include <unistd.h>
+#elif OS(QNX)
+#include <sys/syspage.h>
 #elif OS(WINDOWS)
 #include <windows.h>
 #endif
@@ -70,6 +72,9 @@ int numberOfProcessorCores()
     long sysconfResult = sysconf(_SC_NPROCESSORS_ONLN);
 
     s_numberOfCores = sysconfResult < 0 ? defaultIfUnavailable : static_cast<int>(sysconfResult);
+#elif OS(QNX)
+    int numCpuQNX = _syspage_ptr->num_cpu;
+    s_numberOfCores = numCpuQNX < 0 ? defaultIfUnavailable : numCpuQNX;
 #elif OS(WINDOWS)
     UNUSED_PARAM(defaultIfUnavailable);
     SYSTEM_INFO sysInfo;
diff --git a/Source/WTF/wtf/PlatformHave.h b/Source/WTF/wtf/PlatformHave.h
index 0830d313..5cd2fb96 100644
--- a/Source/WTF/wtf/PlatformHave.h
+++ b/Source/WTF/wtf/PlatformHave.h
@@ -211,7 +211,7 @@
 #define HAVE_MADV_FREE_REUSE 1
 #endif
 
-#if OS(DARWIN) || OS(HAIKU)
+#if OS(DARWIN) || OS(HAIKU) || OS(QNX)
 #define HAVE_MADV_DONTNEED 1
 #endif
 
@@ -411,7 +411,7 @@
 #define HAVE_URL_FORMATTING 1
 #endif
 
-#if !OS(WINDOWS)
+#if !OS(WINDOWS) && !OS(QNX)
 #define HAVE_STACK_BOUNDS_FOR_NEW_THREAD 1
 #endif
 
diff --git a/Source/WTF/wtf/StackBounds.cpp b/Source/WTF/wtf/StackBounds.cpp
index 5759cb2b..ada2ac38 100644
--- a/Source/WTF/wtf/StackBounds.cpp
+++ b/Source/WTF/wtf/StackBounds.cpp
@@ -81,29 +81,29 @@ WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
     return newThreadStackBounds(pthread_self());
 }
 
-#elif OS(UNIX) || OS(HAIKU)
-
-#if OS(OPENBSD)
+#elif OS(QNX)
 
-StackBounds StackBounds::newThreadStackBounds(PlatformThreadHandle thread)
+StackBounds StackBounds::currentThreadStackBoundsInternal()
 {
-    stack_t stack;
-    pthread_stackseg_np(thread, &stack);
-    void* origin = stack.ss_sp;
+    struct _thread_local_storage* tls = __tls();
+    void* bound = tls->__stackaddr;
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-    void* bound = static_cast<char*>(origin) - stack.ss_size;
+    void* origin = static_cast<char*>(bound) + tls->__stacksize;
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
     return StackBounds { origin, bound };
 }
 
-#elif OS(QNX)
+#elif OS(UNIX) || OS(HAIKU)
+
+#if OS(OPENBSD)
 
 StackBounds StackBounds::newThreadStackBounds(PlatformThreadHandle thread)
 {
-    struct _thread_local_storage* tls = __tls();
-    void* bound = tls->__stackaddr;
+    stack_t stack;
+    pthread_stackseg_np(thread, &stack);
+    void* origin = stack.ss_sp;
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-    void* origin = static_cast<char*>(bound) + tls->__stacksize;
+    void* bound = static_cast<char*>(origin) - stack.ss_size;
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
     return StackBounds { origin, bound };
 }
diff --git a/Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp b/Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp
index f05da3d7..e5d07507 100644
--- a/Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp
+++ b/Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp
@@ -221,7 +221,7 @@ void* OSAllocator::reserveAndCommit(size_t bytes, Usage usage, void* address, bo
 
 void OSAllocator::commit(void* address, size_t bytes, bool writable, bool executable)
 {
-#if OS(LINUX) || OS(HAIKU)
+#if OS(LINUX) || OS(HAIKU) || OS(QNX)
     UNUSED_PARAM(writable);
     UNUSED_PARAM(executable);
     while (madvise(address, bytes, MADV_WILLNEED) == -1 && errno == EAGAIN) { }
